### PR TITLE
requireSemicolons: Don't require semicolon for class and function exports (fixes #1517)

### LINKED ISSUE #1517: requireSemicolons should not warn on exporting functions or classes
references: #1484 and 88cdfa65ed15ecf0e000834352e9895a73b797ab

This code should be valid with `requireSemicolons: true`:

```js
export class MyClass {
}
export default class MyClass {
}
export function myFunction() {
}
export default function myFunction() {
}
```

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/23992901-requiresemicolons-should-not-warn-on-exporting-functions-or-classes?utm_campaign=plugin&utm_content=tracker%2F281640&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F281640&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by lahmatiy**

### CODE REVIEW

**Comment by lahmatiy** [comment]
All tests passed but something wrong with build process :(

File: test/specs/rules/require-semicolons.js [hunk 1]
```diff
@@ -237,23 +237,20 @@ describe('rules/require-semicolons', function() {
...
+            'export default class MyClass {}'
         ]);
 
         invalid([
             'export default foo',
             'export default foo//;',
             'export default foo\n//;',
-            'export function foo(){}',
-            'export function foo(){}//;',
-            'export function foo(){}\n//;',
```
Comment (zxqfox) [inline test/specs/rules/require-semicolons.js]: do we need these checks with `//;` at the end of line?

**Comment by markelog** [comment]
> All tests passed but something wrong with build process :(

Yeah, sorry about that, coveralls started to do this a lot, build restarted, all green now

**Comment by lahmatiy** [comment]
> do we need these checks with //; at the end of line?

@zxqfox Yeah, I think those test should be removed as more parser relevant tests. It would be great if somebody do it (I have no time right now).

**Comment by hzoo** [comment]
Do we want to try another code coverage service if there is one?

**Comment by lahmatiy** [comment]
@zxqfox I removed some redundant tests.

**Comment by markelog** [comment]
Thank you!

**Pull request closed without merging**